finance_dl/amazon.py

```diff
@@ -147,12 +154,20 @@ def login(self):
         logger.info('Logged in')
         self.logged_in = True
 
-    def get_invoice_path(self, order_id):
+    def get_invoice_path(self, year, order_id):
+        if self.dir_per_year:
+            return os.path.join(self.output_directory, year, order_id + '.html')
         return os.path.join(self.output_directory, order_id + '.html')
 
     def get_orders(self, regular=True, digital=True):
         invoice_hrefs = []
         order_ids_seen = set()
+        order_ids_downloaded = frozenset([
+            name[:len(name)-5]
+            for _, _, files in os.walk(self.output_directory)
```

Comment:
Perhaps replace with something like
`order_ids_downloaded = [os.path.basename(f).replace('.html', '') for f in glob.glob(os.path.join(self.output_directory, "*.html"))]`

Reply:
Eh, not that much shorter.

Reply:
Tried this a couple different ways and in this spot I think it makes readability worse. The need to add the extra basename call, plus the os.path.join to construct the argument to glob, plus the extra required `recursive` argument to glob, add quite a bit of noise obscuring what's otherwise fairly clear intent.